### PR TITLE
Implement TTL for deferred jobs

### PR DESCRIPTION
**Pull request by hberntsen**
This builds upon #1685 and fixes the problem that jobs can stay forever in the deferred queue.

### CODE REVIEW

**Comment by anton-daneyko-ultramarin** [comment]
@hberntsen Thanks for tackling this. This is an issue for me ATM.

Maybe you have an idea if it is possible to set the status of the child jobs to failed if the parent job failed?

File: rq/registry.py [hunk 1]
```diff
@@ -398,11 +398,45 @@ class DeferredJobRegistry(BaseRegistry):
...
+
+        if job_ids:
+            failed_job_registry = FailedJobRegistry(self.name, self.connection, serializer=self.serializer)
+
+            with self.connection.pipeline() as pipeline:
+                for job_id in job_ids:
+                    try:
+                        job = self.job_class.fetch(job_id, connection=self.connection, serializer=self.serializer)
+                    except NoSuchJobError:
+                        continue
```
Comment (selwin) [inline rq/registry.py:418]: Mind adding a little test for this line so test coverage doesn't decrease?

File: rq/registry.py [hunk 2]
```diff
@@ -398,11 +398,45 @@ class DeferredJobRegistry(BaseRegistry):
...
+
+                    job.set_status(JobStatus.FAILED)
```
Comment (selwin) [inline rq/registry.py:420]: Do you mine passing in pipeline in `set_status`?

**Comment by hberntsen** [comment]
@anton-daneyko-ultramarin I'd need to look into that. It's probably something I'll also need myself

**Comment by hberntsen** [comment]
@selwin Done! I've done some additional code cleanup as well

**Comment by selwin** [comment]
Thanks!

**Pull request merged**